cabal.project

```diff
@@ -52,5 +52,236 @@ source-repository-package
 
 source-repository-package
   type: git
-  location: https://github.com/michaelpj/slack-web.git
-  tag: 6ff88895c5da593e7bd09a46d7002441817aa6f8
+  location: https://github.com/shmish111/github.git
+  tag: cc27b9de4d5d0939235fa9a8b418de3ea4807bab
+
+-- Node protocols
+
+-- We need to relax the dependency on GHC 8.6 for the
+-- ouroboros-network* packages (and hashtable)
+allow-newer: Win32-network:base
+           , github:base
+           , goblins:base
+           , io-sim:base
+           , io-sim-classes:base
+           , network-mux:base
+           , ouroboros-network:base
+           , ouroboros-network-framework:base
+           , cardano-binary:base
+           , ouroboros-network:hashable
+
+           -- The following two dependencies are needed by plutus.
+           , eventful-sql-common:persistent
+           , eventful-sql-common:persistent-template
+
+-- Protolude gets updated to 0.3 when using a newer
+-- version of GHC, not sure why.
+constraints:
+  protolude == 0.2.4
+
+-- This is also needed so evenful-sql-common will build with a
```

Comment:
I am not sure why or how this works currently with cabal. Is it possible that for a cabal build the old version of `persistent` is used?
@krisajenkins I think you did those flag modifications in `stack.yaml`, how come you don't need those in the `cabal.project` file?

Reply:
I think I did those changes. I should have fixed this at the time, sorry!

I'm actually also confused as to how this works currently. I *think* what's actually happening is:
- stack
    - Has a particular version of persistent in the snapshot, so it forces that.
    - We therefore fixup `eventful-sql-common` to accept that version via the extra flags trick.
- cabal
    - Instead *solves* for an older version of persistent based on the constraint from `eventful-sql-common`, thus not requiring the flags.

This makes me think that maybe we could solve our problem for stack also by putting an earlier version of `persistent` as an `extra-dep` instead.

I can look into this separately.

Reply:
Do put the bits relating to `eventful-sql-common` outside the section for the node protocols, though!

Reply:
I've read that cabal can use different versions of the same library as a dependency for different packages and it should work out (as long as those packages don't export any APIs from the conflicting dependencies).